tests/base.py

```diff
@@ -44,6 +45,34 @@ def has_roce_hw_bug(vendor_id, vendor_part_id):
 
 
 class PyverbsAPITestCase(unittest.TestCase):
+    """
+    A base class for test cases which provides the option for user parameters.
+    These can be provided by manually adding the test case to the runner:
+    suite = unittest.TestSuite()
+    ... # Regular auto-detection of test cases, no parameters used.
+    # Now follows your manual addition of test cases e.g:
+    suite.addTest(PyverbsAPITestCase.parametrize(<TestCaseName>, dev_name='..'))
+    """
+    def __init__(self, methodName='runTest', dev_name=None):
+        super(PyverbsAPITestCase, self).__init__(methodName)
+        if parser.args['dev_name']:
+            self.dev_name = parser.args['dev_name']
+        else:
+            self.dev_name = dev_name
+
+    @staticmethod
+    def parametrize(testcase_klass, dev_name=None):
```

Comment:
Is this ever called? Why is it being added?

Reply:
This was added to give the user an API to call parametrize() when creating a TestSuite like what is done in RDMATestCase, but I decided to drop it in v2 of this patch set.